bin/github-review.sh

```diff
@@ -45,53 +41,130 @@ if ! [[ $1 =~ ^[0-9]+$ ]]; then
+edgeCase=0
+echo "$fileLineChangeJSON" | jq -r '.files[] | "\(.path) \(.additions) \(.deletions)"' | while read -r line; do
+    path=$(echo $line | cut -d ' ' -f1)
+    additions=$(echo $line | cut -d ' ' -f2)
+    deletions=$(echo $line | cut -d ' ' -f3)
+
+    # Now you can perform checks on $additions and $deletions as per your requirements
+    if [[ ($additions == 2 && $deletions == 1) ]]; then
+        edgeCase=1
+    elif [[ ($additions != 1 || $deletions != 0) ]]; then
+        echo "ERROR: $path has invalid changes" 1>&2
+        exit 1
+    fi
+done
+
+if [[ $edgeCase == 1 ]]; then
+    echo "Edge case when adding chain with the highest chain id number"
+    diffPatchSeparated=($(gh pr diff $pr | grep -E '^[+-] '))
+    # Adding three elements at a time together to compare from the output array
+    diffPatch=()
+    for ((i=0; i<${#diffPatchSeparated[@]}; i+=3)); do
+        diffPatch+=("${diffPatchSeparated[i]} ${diffPatchSeparated[i+1]} ${diffPatchSeparated[i+2]}")
+    done
+
+    # Initialize a counter
+    counter=1
+    # Initialize a flag to indicate if the previous pattern was correct
+    pattern_correct=true
+    # Read input line by line from the array
+    for line in "${diffPatch[@]}"; do
+        # Determine the line type based on the counter
+        case $((counter % 3)) in
+            1) # First line should start with '-'
+                if [[ $line != -* ]]; then
+                    pattern_correct=false
+                fi
+                # Store the first line to compare with the second line
+                first_line="$line"
+                ;;
+            2) # Second line should be the first line with a comma
+                expected_line="+${first_line:1},"
+                if [[ $line != $expected_line ]]; then
+                    pattern_correct=false
+                fi
+                ;;
+            0) # Third line should start with '+'
+                if [[ $line != +* ]]; then
+                    pattern_correct=false
+                fi
+                # Check the pattern for the set of three lines
+                if [ "$pattern_correct" = false ]; then
+                    echo "Unknown lines added or removed" 1>&2
+                    exit 1
+                fi
+                # Reset the pattern_correct flag for the next set of lines
+                pattern_correct=true
+                ;;
+        esac
+        # Increment the counter
+        ((counter++))
+    done
+fi
+
 echo "Verifying Deployment Asset"
-gh pr diff $pr --patch | git apply --include 'src/assets/**'
+gh pr diff $pr --patch | git apply --include 'src/assets/v'$version'/**'
 
 # Getting default addresses, address on the chain and checking code hash.
+deploymentTypes=($(jq -r --arg c "$chainid" '.networkAddresses[$c]' "$versionFiles"))
+if [[ $deploymentTypes == "[" ]]; then
+    deploymentTypes=($(jq -r --arg c "$chainid" '.networkAddresses[$c][]' "$versionFiles"))
+fi
```

Comment:
IMO this is easier to express in `jq`:

```suggestion
deploymentTypes=($(jq -r --arg c "$chainid" '[.networkAddresses[$c]] | flatten | .[]' "$versionFiles"))
```

Reply:
Updated, lot of new things to learn with this bash script ❤️